rust_src/src/editfns.rs

```diff
@@ -341,4 +347,242 @@ pub fn string_to_char(string: LispStringRef) -> EmacsInt {
     }
 }
 
+#[lisp_fn(min = "0")]
+pub fn line_beginning_position(n: Option<EmacsInt>) -> EmacsInt {
+    let mut charpos: isize = 0;
+
+    let n = n.unwrap_or(1) as isize;
+
+    unsafe { scan_newline_from_point(n - 1, &mut charpos, ptr::null_mut()) };
+
+    // Return END constrained to the current input field.
+    constrain_to_field(
+        Some(LispNumber::Fixnum(charpos as EmacsInt)),
+        LispNumber::Fixnum(point() as EmacsInt),
+        n != 1,
+        true,
+        LispObject::constant_nil(),
+    )
+}
+
+#[lisp_fn(min = "0")]
+pub fn line_end_position(n: Option<EmacsInt>) -> EmacsInt {
+    let orig = point();
+
+    let n = n.unwrap_or(1);
+
+    let clipped_n = clip_to_bounds(ptrdiff_t::min_value() + 1, n, ptrdiff_t::max_value());
+    let end_pos = unsafe {
+        find_before_next_newline(
+            orig as isize,
+            0,
+            clipped_n - (if clipped_n <= 0 { 1 } else { 0 }),
+            ptr::null_mut(),
+        )
+    };
+
+    // Return END constrained to the current input field.
+    constrain_to_field(
+        Some(LispNumber::Fixnum(end_pos as EmacsInt)),
+        LispNumber::Fixnum(orig),
+        n != 1,
+        true,
+        LispObject::constant_nil(),
+    )
+}
+
+#[lisp_fn(min = "0")]
```

Comment:
You're missing docstrings on a couple of these.

Reply:
Oh. I was switching functions during debugging and must have neglected to return the comments. Easy fix.